### PR TITLE
Add pop of color to terminal output with a color formatter

### PR DESCRIPTION
**Pull request by bobcatfish**
This implementation avoids needing to instantiate a formatter object
(as seen in #850) by using a global function pointer to override the
IsTerminal logic. This means any code at any time could output in
whatever color it wants to. (Thanks @r2d4 for suggesting this
implementation in the first place!)

Skaffold draws a distinction between log output, which goes to stdout,
and output from Skaffold to the user, which goes to stdout and is
interleaved with output from the tools Skaffold users (e.g. docker for
building). This commit changes the output from Skaffold to the user to
be Blue!

The escape codes to make the text blue (or other colors, in the case of
the kube pod-specific logging) will only be output if stdout is a
terminal. If output is being redirected to a file for example, the
escape codes will not be included.

This change makes use of the exciting colors.go logic but splits it
into the picker (specifically for mapping images to colors consistently)
from the logic to add escape codes.

Other alternatives:
* #856 - Wrap io.Writer
* #850 - Instantiate a formatter object
* Separate formatting logic from writing logic (I tried this out and it
  didn't seem like it improved anything as it didn't align well with
  how `fmt` is implemented)

### CODE REVIEW

File: pkg/skaffold/deploy/helm.go [hunk 1]
```diff
@@ -117,7 +118,7 @@ func (h *HelmDeployer) deployRelease(out io.Writer, r v1alpha2.HelmRelease, buil
 		return nil, errors.Wrap(err, "cannot parse the release name template")
 	}
 	if err := h.helm(out, "get", releaseName); err != nil {
-		fmt.Fprintf(out, "Helm release %s not installed. Installing...\n", releaseName)
+		color.Fprintf(out, color.Red, "Helm release %s not installed. Installing...\n", releaseName)
```
Comment (bobcatfish) [inline pkg/skaffold/deploy/helm.go]: I made this red just for fun!

File: pkg/skaffold/color/colors.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
+/*
```
[the remark this answers is not in the document]
Comment (bobcatfish) [reply pkg/skaffold/color/colors.go]: I'm guessing you want to do this b/c the color functionality isn't really a "core part" of skaffold, it's like an extra utility, and I see what you're saying, but I find that calling packages `util` or `lib` or anything like that b/c it makes it very hard to reason about what should and should not be inside them. I'd be more inclined to take the contents of `pkg/skaffold/util/` and move them into their own packages.

See "bad package names" at https://blog.golang.org/package-names

File: pkg/skaffold/color/colors.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+	"fmt"
+)
+
+// Color can be used to format text using ANSI escape codes so it can be printed to
+// the terminal in color.
+type Color int
+
+var (
+	// LightRed can format text to be displayed to the terminal in light red, using ANSI escape codes.
+	LightRed = Color(91)
```
[the remark this answers is not in the document]
Comment (bobcatfish) [reply pkg/skaffold/color/colors.go]: The other file _looks_ like it's duplicating this but actually it is only specifying the order of the colors to use when coloring pod output

File: pkg/skaffold/color/colors.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (bobcatfish) [reply pkg/skaffold/color/colors.go]: But also it's not great that the `Color` type is `color.Color` :(

File: cmd/skaffold/app/cmd/fix.go [hunk 4]
```diff
@@ -48,7 +48,7 @@ func NewCmdFix(out io.Writer) *cobra.Command {
 				return
 			}
 			if cfg.GetVersion() == config.LatestVersion {
-				fmt.Fprintln(out, "config is already latest version")
+				color.Fprintln(out, color.Default, "config is already latest version")
```
[the remark this answers is not in the document]
Comment (bobcatfish) [reply cmd/skaffold/app/cmd/fix.go]: ooooo that's a cool idea

**Comment by bobcatfish** [comment]
Made the change you suggested @dgageot !

**Comment by dgageot** [comment]
LGTM!

**Pull request merged**